cdap-security/src/main/java/io/cdap/cdap/security/auth/AbstractKeyManager.java

```diff
@@ -118,16 +117,34 @@ protected final KeyIdentifier generateKey() {
       nextId = rand.nextInt(Integer.MAX_VALUE);
```

Comment:
Does the `keyId` need to be a secure value? If so, would recommend using [SecureRandom](https://docs.oracle.com/javase/8/docs/api/java/security/SecureRandom.html) instead of Random, otherwise this value can be guessed.

Reply:
No it doesn’t

Reply:
Acknowledged, thanks!